Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm

```diff
@@ -367,6 +373,53 @@
     protectedThrottler()->setShouldTakeNearSuspendedAssertion(false);
 }
 
+RefPtr<XPCEventHandler> WebProcessProxy::xpcEventHandler() const
+{
+    return adoptRef(new WebProcessProxy::XPCEventHandler(*this));
+}
+
+bool WebProcessProxy::XPCEventHandler::handleXPCEvent(xpc_object_t event) const
+{
+#if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
+    auto messageName = xpcDictionaryGetString(event, XPCEndpoint::xpcMessageNameKey);
+    if (messageName == logMessageName) {
+        RefPtr webProcess = m_webProcess.get();
+        if (!webProcess)
+            return true;
+
+        MESSAGE_CHECK_WITH_RETURN_VALUE_BASE(webProcess->m_logEndpointEnabled, webProcess->connection(), false);
+
+        auto subsystem = xpcDictionaryGetString(event, subsystemKey);
+        auto category = xpcDictionaryGetString(event, categoryKey);
+        auto messageString = xpcDictionaryGetString(event, messageStringKey);
+        auto logType = xpc_dictionary_get_uint64(event, logTypeKey);
+        auto pid = xpc_connection_get_pid(xpc_dictionary_get_remote_connection(event));
+
+        OSObjectPtr<os_log_t> osLog;
+        if (!subsystem.isEmpty() && !category.isEmpty())
+            osLog = adoptOSObject(os_log_create(subsystem.utf8().data(), category.utf8().data()));
+
+        auto osLogPointer = osLog ? osLog.get() : OS_LOG_DEFAULT;
+        os_log_with_type(osLogPointer, static_cast<os_log_type_t>(logType), "WebContent[%d] %{public}s", static_cast<int>(pid), messageString.utf8().data());
+        ++webProcess->m_logCountForTesting;
```

Comment:
Can we make this name less generic? maybe `m_logCountDuringLaunchForTesting`? Also, given how this is used, maybe this can simply be a boolean: `bool m_didReceiveLogsDuringLaunchForTesting`